certora/specs/SocialRecoveryModule.spec

```diff
@@ -73,104 +37,95 @@ function requireSocialRecoveryModuleEnabled() {
     require(safeContract.isModuleEnabled(currentContract));
 }
 
-// A helper function to verify that two addresses are not the same.
-function notSameAddress(address a, address b) {
-    require(a != b);
-}
-
-// This rule verifies that the addGuardianWithThreshold(...) works as expected.
+// This integrity rule verifies that the addGuardianWithThreshold(...) works as expected.
 rule addGuardianWorksAsExpected(env e, address guardian, uint256 threshold, address otherGuardian) {
 
     // This can be removed once linked list invariant is implemented.
-    require ghostGuardianCount[safeContract] > 0 => ghostGuardians[safeContract][SENTINEL()] != 0; // Sentinel should not be zero if there are guardians.
+    require guardianStorageContract.entries[safeContract].count > 0 => guardianStorageContract.entries[safeContract].guardians[SENTINEL()] != 0; // Sentinel should not be zero if there are guardians.
 
-    notSameAddress(guardian, otherGuardian); // Should not be the same as the other guardian.
+    require guardian != otherGuardian;
```

Comment:
Can we explain why this is needed? Naively, I would expect the `     currentContract.addGuardianWithThreshold` to revert if `guardian == otherGuardian`.

Reply:
The problem here might be that the `otherGuardian` can actually be not a guardian, and be the same address as `guardian`, which will change the state.

Will remove the non-equivalence, and add a requirement that the `otherGuardian` is a guardian.

Reply:
> add a requirement that the otherGuardian is a guardian.

Hmm, I wouldn't. As is it has a nice property that `isGuardian(otherAccount)` never changes to a call of `addGuardian(...)` with guardian.

Can we just rename this to `address otherAccount`, I think that will be much clearer.